fretboard/src/main/java/mozilla/components/service/fretboard/JSONExperimentParser.kt

```diff
@@ -63,6 +66,36 @@ class JSONExperimentParser {
         jsonObject.putIfNotNull(DESCRIPTION_KEY, experiment.description)
         jsonObject.put(ID_KEY, experiment.id)
         jsonObject.putIfNotNull(LAST_MODIFIED_KEY, experiment.lastModified)
+        jsonObject.putIfNotNull(PAYLOAD_KEY, payloadToJson(experiment.payload))
+        return jsonObject
+    }
+
+    private fun jsonToPayload(jsonObject: JSONObject): ExperimentPayload {
+        val payload = ExperimentPayload()
+        for (key in jsonObject.keys()) {
+            val value = jsonObject.get(key)
+            if (value !is JSONObject) {
+                when (value) {
+                    is JSONArray -> payload.put(key, value.toList<Any>())
+                    else -> payload.put(key, value)
+                }
+            }
+        }
+        return payload
+    }
```

Comment:
> since I thought the metadata passed to an experiment would be pretty basic (primitive values and lists of primitive values), correct me if that's not the case :)

We can start with that. But this may change. Lets mention in a comment here that for now we decided to only support primitive values.

Reply:
That's fine. Well, actually the way this is designed it could allow any complex object to be added to the ExperimentPayload class, even a JSONObject if that's the right need, or even a List (although it's more oriented to primitive types and for getting a list of an unsupported type it would need a cast)